aip/0152.md

```diff
@@ -0,0 +1,84 @@
+---
+aip:
+  id: 152
+  state: reviewing
+  created: 2020-04-27
+permalink: /152
+redirect_from:
+  - /0152
+---
+
+# Jobs
+
+Occasionally, APIs may need to expose a task that takes significant time to
+complete, and where a transient [long-running operation][aip-151] is not
+appropriate. For example, a task could need to run repeatedly, or have separate
+permissions for configuring the task as opposed to running it.
+
+## Guidance
+
+An API **may** define a `Job` resource to represent a particular task with
+distinct setup, configuration, and execution:
+
+```proto
+message WritingJob {
+  // The resource name for the writing job.
+  string name = 1 [(google.api.resource) = {
+    type: "library.googleapis.com/WritingJob"
+    pattern: "publishers/{publisher}/writingJobs/{writing_job}"
+  }];
+
+  // Additional configuration...
+}
+```
+
+- The name of the resource **must** end with the word "Job".
+- The service **should** define all five of the standard methods, and use them
+  as the primary way to configure the job.
+
+### Run method
+
+The service **must** define a `Run` custom method that executes the job:
+
+```proto
+rpc RunWritingJob(RunWritingJobRequest)
+    returns (google.longrunning.Operation) {
+  option (google.api.http) = {
+    post: "/v1/{name=publishers/*/writingJobs/*}:run"
+    body: "*"
+  }
+  option (google.longrunning.operation_info) = {
+    response_type: "RunWritingJobResponse"
+    metadata_type: "RunWritingJobMetadata"
+  }
+}
+```
+
+- The HTTP method **must** be `POST`.
+- The method **must** return a [long-running operation][aip-151], which
+  **must** resolve to a response message that includes the result of running
+  the job.
+  - The method **may** use any metadata message it wishes.
+- Errors that prevent execution of the job from _starting_ **must** return an
+  error response ([AIP-193][]), similar to any other method. Errors that occur
+  over the course of the job execution **may** be placed in the metadata
+  message. The errors themselves **must** still be represented with a
+  [`google.rpc.Status`][] object.
+
+### Results
+
+Ordinarily, the API **should** provide results to the user as the final
+response of the `Run` method. However, this is sometimes insufficient; for
+example, a job that runs on a cadence in the background can not deliver results
+to the user in this way.
+
+The service **may** store results as a sub-collection of resources under the
```

Comment:
I think this needs a lot more explanation. What should we call the subcollection? results? runs? Can we see an example? We need to expand on this a lot.